actors/evm/src/interpreter/instructions/storage.rs

```diff
@@ -29,10 +25,6 @@ pub fn sstore(
         return Err(StatusCode::StaticModeViolation);
     }
 
-    let location = state.stack.pop();
-    let value = state.stack.pop();
-    let opt_value = if value == U256::zero() { None } else { Some(value) };
-
-    system.set_storage(location, opt_value)?;
+    system.set_storage(state.stack.pop(), state.stack.pop())?;
```

Comment:
Wait, this is potential undefined behavior with the pops.
Does rust mandate ledt to right evaluation order for args?
In any othe language you've shot yourself in the foot here.

Reply:
left -> right, in -> out. Yes.

Reply:
uhm ok, interesting.
Can we still either pop first to named or add a comment regardless?

Reply:
Sure, we can pop first. I agree that it does make the pop order clearer.

Reply:
left to right is how it is evaluated https://doc.rust-lang.org/reference/expressions.html#evaluation-order-of-operands

Reply:
I have to say, it is _highly_ unusual of high perf languages to do that.

The compiler is allowed to go bersek here, ocaml does right to left, some C compilers do whatever works faster situationally, and in scheme you are exoelled if you assume evaluation order :)